userland/src/add_two_ints_client.cpp

```diff
@@ -0,0 +1,43 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <sys/time.h>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <simple_msgs/AllBuiltinTypes.h>
+#include <simple_msgs/AllDynamicArrayTypes.h>
+#include <simple_msgs/AllPrimitiveTypes.h>
+#include <simple_msgs/AllStaticArrayTypes.h>
+#include <simple_msgs/Nested.h>
+#include <simple_msgs/String.h>
+#include <simple_msgs/Uint32.h>
+
+#include <userland_msgs/AddTwoInts.h>
+
+int main(int argc, char** argv)
+{
+  rclcpp::init(argc, argv);
+
+  auto node = rclcpp::Node::make_shared("add_two_ints_client");
+
+  auto client = node->create_client<userland_msgs::AddTwoInts>("add_two_ints");
+  auto request = std::make_shared<userland_msgs::AddTwoInts::Request>();
+  request->a = 2;
+  request->b = 3;
+
+  try
+  {
+    auto response = client->send_request(request);
```

Comment:
Since this signature implies allocation of the response we should implement a different signature on the lower level:

    rc = client->send_request(request, response);

This way the response can be preallocated by the user land code.
The previous signature can be provided as syntactic sugar on-top.